### DISCUSSION TITLE
Can I have multiple myenergi API keys?

### DISCUSSION THREAD

**Discussion by t0mb080** [CJNE/ha-myenergi#539] [General]
Just trying to set up the integration and have a bit of a beginner question...

I currently have Octopus Intelligent controlling my Zappi charger which uses an existing myenergi API key by the look of it.  To add the HA myenergi integration, can I safely generate a new API key to use?  I just want to check that it won't mess up my existing Octopus Intelligent control before hitting the button...

Thanks very much for any help you can give.

**Comment by G6EJD** [comment]
Only one key, just re-use it!

**Comment by t0mb080** [comment]
Thanks very much for the quick reply!

My question now is, how do I find my existing key?  Having clicked around, I can't see it in the myenergi webpage or app?

**Comment by G6EJD** [comment]
You’ll need to generate a new one, but this time make a note of it. Just place it in a file called ME API Key
You’ll need to redo your Octopus integration too.

**Comment by bdlhome** [comment]
Having recently added the integration with Zappi/Eddi & Libbi devices and having to generate a new API Key I had similar questions about Octopus Intelligence Zappi connection.

I too wondered if the New API key may cause problems or need me to redo the Octopus Intelligence, as I couldn't remember if it used the MyEnergi API Key.  My findings are it appears it doesn't use the API key and Myenergi Zappi control has continued to work without issues. 

I thought I would post in case it was useful for any future users searching for answers as I did.
